source/opt/module.cpp

```diff
@@ -137,10 +137,54 @@ void Module::ToBinary(std::vector<uint32_t>* binary, bool skip_nop) const {
   binary->push_back(header_.bound);
   binary->push_back(header_.reserved);
 
-  auto write_inst = [binary, skip_nop](const Instruction* i) {
-    if (!(skip_nop && i->IsNop())) i->ToBinaryWithoutAttachedDebugInsts(binary);
-  };
-  ForEachInst(write_inst, true);
+  if (ext_inst_debuginfo_.empty()) {
```

Comment:
For simplicity, I would prefer to just have one version of the lambda that writes an instruction.  We can see it takes up too much time, then we can reconsider.